packages/compiler-sfc/src/parse.ts

```diff
@@ -95,11 +96,18 @@ export function parse(
     sourceRoot = '',
     pad = false,
     ignoreEmpty = true,
-    compiler = CompilerDOM
+    compiler = CompilerDOM,
+    defaultScriptLang
   }: SFCParseOptions = {}
 ): SFCParseResult {
   const sourceKey =
-    source + sourceMap + filename + sourceRoot + pad + compiler.parse
+    source +
+    sourceMap +
+    filename +
+    sourceRoot +
+    pad +
+    compiler.parse +
+    (defaultScriptLang != null ? '@' + defaultScriptLang : '')
```

Comment:
why?

Reply:
It's for keeping the compatibility to the old version.

I think if it makes the script lang option necessary, the script compiler might be broken.

Reply:
Well, I made `<script>` src required just now.